### DISCUSSION TITLE
Definitions of timestamps in alerts

### DISCUSSION THREAD

**Discussion by sdtaylor** [weather-gov/api#590] [Q&A] [answered]
Hi, similar to #385, what are the definitions for the other timestamps in the alerts api? Specifically sent, effective, and onset? 

```
                "sent": "2021-12-30T03:04:00-06:00",
                "effective": "2021-12-30T03:04:00-06:00",
                "onset": "2022-01-01T06:00:00-06:00",
                "expires": "2021-12-30T11:15:00-06:00",
                "ends": "2022-01-02T00:00:00-06:00",
```

**Comment by JJKraw** [comment]
This is just from observation over months of reading the API alerts. Corrections / clarifications encouraged.

Sent is simply what it means. The time the particular product was issued.

Effective is when the watch / warning / etc. goes into effect. From what I've seen, it mostly equals the sent time, but I suppose in the case where lead times are important, an alert will be sent before it goes into effect.

Onset is when the conditions described by the product will begin. For instance, there is a freeze warning in Alabama that was issued this morning and the onset time is for 9pm tonight.

Expires refers to the product rather than the event. The particular text in the product will no longer be valid and, if the event is still ongoing, a new product will be issued before the expiration time.

Ends refers to the conditions described by the product - when the watch / warning / etc. will end.

**Reply by sramsay** [reply]
It's the "expires" field that is confusing me.  Right now, the API is listing a storm watch in my area as having expired two hours ago, but there's no new alert.  That would be fine, if every web app (including the NWS's own page) weren't reporting the alert as if that field had no meaning.

Intuitively, it seems like an expired alert should not be displayed.

**Reply by JJKraw** [reply]
I think in that case it's human error. A new alert should have been issued to replace the expired alert if the event is still ongoing.

**Reply by sramsay** [reply]
Good!

Well, not good . . . but you know what I mean. ;) I originally had the problem of my program showing expired alerts, and I thought I had fixed it.  I think there's still a bug in my system, though.

I am the author of nowa: https://github.com/sramsay/nowa

**Reply by sullynole** [reply]
CAP v1.2 documentation can be found here, which should help with understanding the expiration times: https://vlab.noaa.gov/web/nws-common-alerting-protocol/cap-documentation

**Comment by sullynole** [comment, marked as answer]
> Hi, similar to #385, what are the definitions for the other timestamps in the alerts api? Specifically sent, effective, and onset? 
> 
> ```
>                 "sent": "2021-12-30T03:04:00-06:00",
>                 "effective": "2021-12-30T03:04:00-06:00",
>                 "onset": "2022-01-01T06:00:00-06:00",
>                 "expires": "2021-12-30T11:15:00-06:00",
>                 "ends": "2022-01-02T00:00:00-06:00",
> ```

This can be found in the CAP 1.2 documentation here: https://vlab.noaa.gov/web/nws-common-alerting-protocol/cap-documentation

**Comment by chaunceygardiner** [comment]
The doc confirms what the OP believes should be the meaning of expires:“This is the time at which the information in the message should be considered stale and no longer used.  The NWS forecaster will update or cancel the alert by the <expires> time.”And, like the OP, I have observed that a follow-up alert is not always issued in time; thus, it’s either have an alert disappear and reappear a little later; or pad the expiration time.On Jan 16, 2023, at 7:16 PM, John Sullivan ***@***.***> wrote:﻿
CAP v1.2 documentation can be found here, which should help with understanding the expiration times: https://vlab.noaa.gov/web/nws-common-alerting-protocol/cap-documentation
